### PR TITLE
feat: Add login screen.

### PR DESCRIPTION
**Pull request by BURG3R5**
- [x] UI
- [x] Bloc
- [x] Tests
- [x] Docs

### CODE REVIEW

File: android/app/build.gradle [hunk 1]
```diff
@@ -26,7 +26,7 @@ apply plugin: 'kotlin-android'
 apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 
 android {
-    compileSdkVersion 30
+    compileSdkVersion 31
```
Comment (praeclarumjj3) [inline android/app/build.gradle:29]: Let's make it 32. `31` was giving me issues.

File: lib/data/config/config.dart [hunk 2]
```diff
@@ -2,7 +2,7 @@ class Environment {
   /// URL that is common to most outgoing requests.
   static const String baseUrl = String.fromEnvironment(
     'BASE_URL',
-    defaultValue: 'https://codephile.mdg.iitr.ac.in/v1/',
+    defaultValue: 'https://codephile-test.herokuapp.com/v1/',
```
Comment (praeclarumjj3) [inline lib/data/config/config.dart:5]: I assume you started a test server to test the login functionality? When I try to log in, it shows `incorrect credentials.`

File: lib/presentation/login/widgets/dialog_and_toast_wrapper.dart [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            Fluttertoast.showToast(msg: message);
+          }
+        }
+      },
+      listenWhen: (state, previousState) =>
+          (state.showDialog != previousState.showDialog) ||
+          (state.status != previousState.status),
+      child: child,
+    );
+  }
```
Comment (praeclarumjj3) [inline lib/presentation/login/widgets/dialog_and_toast_wrapper.dart:40]: I think you can increase the font size and change the font color. The current toast message can be easily overlooked, appearing over the `LOGIN` button with a gray background and white font color.

**Review by praeclarumjj3 — CHANGES_REQUESTED** [review]
- [ ] Add the steps to the readme.
- [ ] Add a `TODO:` statement maybe for changing the initial_route in `main_app.dart`.
- [ ] Look at the dialog box for toast.

File: lib/data/config/config.dart (hunk 2 above)
Comment (BURG3R5) [reply lib/data/config/config.dart:5]: Yeah. This is hitting the debug server, and there's no facility to sign up yet. ('Cause there's no sign up screens yet.)

**Review by praeclarumjj3 — APPROVED** [review]
Looks great, thanks!

**Pull request merged**